src/mcs/EmpiricalDistribution.jl

```diff
@@ -52,6 +52,18 @@ function Statistics.var(d::EmpiricalDistribution)
     return var(d.values, d.weights, corrected=true)
 end
 
+function Base.length(d::EmpiricalDistribution, args::Vararg{Integer,N}) where {N}
+    return length(d.values)
+end
+
+function Base.collect(d::EmpiricalDistribution, args::Vararg{Integer,N}) where {N}
+    return collect(d.values)
+end
+
+function Base.iterate(d::EmpiricalDistribution, args::Vararg{Integer,N}) where {N}
+    return iterate(d.values)
+end
```

Comment:
This strikes me as an incorrect implementation of the iteration interface :) I think you want:
```julia
Base.iterate(x::EmpiricalDistribution, s...) = iterate(x.x, s...)
```